### DISCUSSION TITLE
[Developer Tooling] Textile

### DISCUSSION THREAD

**Discussion by andrewxhill** [filecoin-project/community#89] [Project Showcase]
Hi all! 

Textile is a suite of tools and APIs to help you build on IPFS and Filecoin. Here are some of our projects,

* Powergate - a Filecoin API stack for platform builders: [Powergate](https://github.com/textileio/powergate)
* Buckets - simple object storage on IPFS: [Buckets](https://github.com/textileio/textile)
* Threads - database & data syncing protocol: [Threads](https://github.com/textileio/go-threads) 
* Textile Hub - hosted APIs and infrastructure for all the above: [Signup](https://textile.io), [Docs](https://docs.textile.io), [JS Clients](https://textileio.github.io/js-textile/)

I'll drop our first update below!

**Comment by andrewxhill** [comment]
## Textile Update-  Feb, 2021

Some things we've been working on lately that we'd love feedback on.

### Miner Index

TLDR: A public API that shares real-time information about miners on the network (including known deal history and some curated fields) so you can find great miners fast. We are baking in some other goodies. 

This is coming together really fast, so we should be able to show you something next sprint. In the meantime, our Discussion is open for questions and feedback here: https://github.com/textileio/textile/discussions/496.

### DID identifiers for datasets

TLDR: we build a data synchronization protocol called Threads that pulls together libp2p and IPLD with some event sourcing goodness. We are working now to add an underlying DID support that will open them up to all sorts of new integrations. 

Chime in here: https://github.com/textileio/go-threads/discussions/483

### Bucket Pinning API

tldr: buckets are object storage over IPFS. They allow collaborators to easily manage and synchronize folders of data from their computers and publish that data over IPFS easily. they also have Filecoin integration that allows you to snapshot them on Filecoin easily. Now, we want to provide a Pinning API on every bucket you create. We are looking into making this a way to Pin a CID direct to Filecoin. 

It's a pretty powerful idea. If you are intrigued, jump in here https://github.com/textileio/textile/discussions/499

### A notary for Web3

TLDR: client verification on the Filecoin network is well suited for single addresses at a time. We've defined a flow that will allow developers to apply for a datacap on behalf of all of their users once. It opens up a lot more options for building apps where users create verified deals right away. 

The Discussion is here: https://github.com/textileio/textile/discussions/495

**Comment by andrewxhill** [comment]
## Textile Update-  Mar 8, 2021

See our previous update here https://github.com/filecoin-project/community/discussions/89#discussioncomment-410919

### Miner Index [RELEASED]

V1 is released! https://blog.textile.io/introducing-the-miner-index/ and https://bafzbeic7zldbidfgowimkbayaojcrawdiaxjkhcu26dsd4e4hbl5klcjqa.textile.space/. We're very excited about this tool and hope projects find it useful. 

There are 3 primary commands / API endpoints.

1. Query: filter or sort for miners based on a large assortment of on-chain and off-chain metrics that we collect from many lotus nodes and hundreds of deals.
2. Get: fetch a profile and latest data/telemetry for a single miner by ID.
3. Calculate: provide one or a list of miners plus data size and deal duration parameters to get real-time cost estimates for storage.

![query-get-calculate](https://user-images.githubusercontent.com/370259/110395801-b0010680-8023-11eb-9a7d-f51cf44bdd22.gif)

We got some good feedback from miners about some tweaks to how we are reporting deals that don't complete. Those updates will come over the next few weeks.

### DID identifiers for datasets

So much progress here! [go-did-resolver](https://github.com/textileio/go-did-resolver) is moving along to give threads support for different did types. 

### Bucket Pinning API

[buckets](https://github.com/textileio/go-buckets) have been split into their own repo where we will now extend with a pinning API.

### A notary for Web3

Still coming together, so feel free to add any questions or comments here: https://github.com/textileio/textile/discussions/495

### Other updates

* We released a few updates to the Buckets API, including a new MV command (allows you to move paths on a remote bucket). We've added the mv command to a little node CLI for managing remote buckets. This differs from the standard buck CLI which is meant to work with local directories of data (that you can sync to remote, buckr is primarily an API tool but also the src shows some fun ways to use our Typescript/Javascript cli libs.  
* We are wrapping up our platform IPFS backup/recovery workflow on Filecoin. Still awaiting verification of our FIL wallet address used for the task. We'll be sharing a public report of our setup for other platforms to follow easily.

**Comment by andrewxhill** [comment]
## Textile Update-  Mar 15, 2021

See our previous update here https://github.com/filecoin-project/community/discussions/89?sort=new#discussioncomment-452078

### GR9 Hackathon

> This is huge. If you are working on a project in the Filecoin ecosystem, the [GR9 Hackathon](https://gitcoin.co/hackathon/gr9/onboard) is an opportunity for you to explore new features and potentially get funded for your work. We are partnering with [Protocol Labs](https://protocol.ai/) to give away [$25,000](https://github.com/protocol/grants/blob/main/gitcoin-gr9-hackathon/all-bounties.md#textile-bounties) in grants for building on [Textile's APIs](https://docs.textile.io/) and open-source tools, and another $35,000 is up for grabs for projects using [IPFS](https://ipfs.io/), [Filecoin](https://filecoin.io/), and [Fleek](https://fleek.co/). All prizes will be paid in FIL! Let's take a look at the Textile bounties.

* Blog post: https://blog.textile.io/funding-the-community/
* Bounty links: https://github.com/protocol/grants/blob/main/gitcoin-gr9-hackathon/all-bounties.md#textile-bounties

Our grants include bounties to build dashboards/visualizations of the miner index, integrate Filecoin into new app features, and more.

### GR9 Grants

We've got some awesome projects from the Textile ecosystem participating in the GR9 Grants, give 'em your support https://twitter.com/textileio/status/1370832874341994497

### 5 Minute product tour

We put together a helpful new choose-your-own adventure slide deck where you can check out each of our products and drill into any you wish to know more about. https://twitter.com/textileio/status/1370406547554066437

### Community tweets that make us proud

* https://twitter.com/christroutner/status/1370197704941309954
* https://twitter.com/MemeofDayDApp/status/1370769567153283093

### Bucket Pinning API

[Lots of progress](https://github.com/textileio/textile/discussions/499?sort=new) here (very related to [Thread DID work](https://github.com/textileio/go-threads/discussions/483)).

Quoting @sanderpick from [slack engineering](https://slack.textile.io/)

> Making some progress on exposing a bucket as an IPFS pinning service:

```sh
> ipfs pin remote service ls
mybucket http://127.0.0.1:8000/thread/bafk6chlb2ez6ee7iwjq7fhjsv7wffzea57sa/buckets/bafzbeidkhyny26jte426cd72pwuttyivtge5ijh36vlj45vczzpizwtx7y
```

> Interact with the bucket from IPFS:

```sh
> ipfs pin remote add --service=mybucket --name=mypin --background bafybeig24a3nubcbescvhwb5rlab7v4r676rpqa3w2uulwjd472qt3rtda
CID:    bafybeig24a3nubcbescvhwb5rlab7v4r676rpqa3w2uulwjd472qt3rtda
Name:   mypin
Status: queued
```

> The pin is tracked in the bucket’s metadata:

```sh
> ipfs pin remote service ls
mybucket http://127.0.0.1:8000/thread/bafk6chlb2ez6ee7iwjq7fhjsv7wffzea57sa/buckets/bafzbeidkhyny26jte426cd72pwuttyivtge5ijh36vlj45vczzpizwtx7y
```

> Making some progress on exposing a bucket as an IPFS pinning service:

```sh
> buck get
{
  "thread": "bafk6chlb2ez6ee7iwjq7fhjsv7wffzea57sa",
  "_id": "bafzbeidkhyny26jte426cd72pwuttyivtge5ijh36vlj45vczzpizwtx7y",
  "owner": "did:key:12D3KooWRQkPKVnGLxBhZbbi8zi8QECVZwbSjfYV7mak5vk99T9p",
  "name": "unnamed",
  "version": 1,
  "path": "/ipfs/bafybeibbva2q5u5huwu2l5n25jzgvo2xxxbavi2ex5jujyu6lf7a7iu3ni",
  "metadata": {
    "": {
      "roles": {
        "*": 1,
        "did:key:12D3KooWRQkPKVnGLxBhZbbi8zi8QECVZwbSjfYV7mak5vk99T9p": 3
      },
      "updated_at": 1615792297065949700
    },
    ".textileseed": {
      "roles": {},
      "updated_at": 1615792297065949700
    },
    "bqpesoi5aegy776lugbf7dpadr7h4ztoo": {
      "roles": {},
      "info": {
        "pin": {
          "created": "2021-03-15T07:44:53.074051506Z",
          "pin": {
            "cid": "bafybeig24a3nubcbescvhwb5rlab7v4r676rpqa3w2uulwjd472qt3rtda",
            "name": "mypin",
            "origins": [
              "/ip4/192.168.64.1/tcp/4001/p2p/12D3KooWFSFtCvJCuimAnSF2n2zW67NroPv8Y29w77JN1q9WpJ4L",
              "/ip4/127.0.0.1/tcp/4001/p2p/12D3KooWFSFtCvJCuimAnSF2n2zW67NroPv8Y29w77JN1q9WpJ4L",
              "/ip6/::1/tcp/4001/p2p/12D3KooWFSFtCvJCuimAnSF2n2zW67NroPv8Y29w77JN1q9WpJ4L",
              "/ip4/192.168.64.1/udp/4001/quic/p2p/12D3KooWFSFtCvJCuimAnSF2n2zW67NroPv8Y29w77JN1q9WpJ4L",
              "/ip4/127.0.0.1/udp/4001/quic/p2p/12D3KooWFSFtCvJCuimAnSF2n2zW67NroPv8Y29w77JN1q9WpJ4L",
              "/ip6/::1/udp/4001/quic/p2p/12D3KooWFSFtCvJCuimAnSF2n2zW67NroPv8Y29w77JN1q9WpJ4L",
              "/ip4/75.142.226.110/tcp/11340/p2p/12D3KooWFSFtCvJCuimAnSF2n2zW67NroPv8Y29w77JN1q9WpJ4L",
              "/ip4/75.142.226.110/udp/11340/quic/p2p/12D3KooWFSFtCvJCuimAnSF2n2zW67NroPv8Y29w77JN1q9WpJ4L"
            ]
          },
          "requestid": "bqpesoi5aegy776lugbf7dpadr7h4ztoo",
          "status": "queued"
        }
      },
      "updated_at": 1615794293141921800
    }
  },
  "created_at": 1615792297065949700,
  "updated_at": 1615794676375010600
}
```

**Reply by jennijuju** [reply]
Thank you for updates and textile updates is featured in[ Filecoin Newsletter #14](https://mailchi.mp/1407c088a2b3/filecoin-news-issue-66046)! Im looking forward to more updates in the future!!

**Comment by andrewxhill** [comment]
## Textile Update-  Mar 31, 2021

See our previous update here https://github.com/filecoin-project/community/discussions/89#discussioncomment-485903

### ChainLink Hackathon & GR9 Hackathon Workshops

We gave two workshops that are each available online now. The workshops were meant to give a nice introduction to Filecoin from the client's perspective. They included a tour of some of the exciting interfaces being build, some of the APIs that are available, and and some of the full-node options. Then, they had a 5 steps to storing data on Filecoin with [buckets](https://github.com/textileio/powergate) and a really quick introduction to [Powergate](https://github.com/textileio/go-buckets). The two workshops are 95% the same material, but here are the two links:

**gr9** https://www.youtube.com/watch?v=VWm9KaL0LkE&feature=youtu.be

**chainlink** https://www.youtube.com/watch?v=NdSdsTsh0QI

### Thread DIDs

The [discussion on thread dids](https://github.com/textileio/go-threads/discussions/483) is attracting more comments, which is exciting. Some good feedback and ideas are bubbling up. 

Carson [submitted a paper](https://twitter.com/carsonfarmer/status/1377368965765206027) to DI2F related to the topic.

### Gr9 Community Support

We were too late this year to be any kind of matching sponsor in the [Gitcoin GR9 Grants](https://gitcoin.co/grants/) round. However, we still wanted to contributed to the Textile community taking part. We set aside 5000DAI. So as to influence other's contributions as little as possible, we held off until the final day. We then voted internally on the set of Textile community projects in the grant round and distributed the DAI according to our internal vote. Here is what we ended up supporting:

| project | funding |
|--------|-------|
| [AstralDao](https://gitcoin.co/grants/1102/astral) | 147 D |
| [Distributed Town](https://gitcoin.co/grants/1485/distributedtown) | 441 D |
| [Endowl](https://gitcoin.co/grants/2074/crypto-inheritance-by-endowl) | 441 D |
| [GeoWeb](https://gitcoin.co/grants/1403/geo-web) | 147 D |
| [Huddle01](https://gitcoin.co/grants/1298/huddle-01-revolutionizing-video-conferences-using) | 735 D |
| [Meme of the Day](https://gitcoin.co/grants/1413/meme-of-the-day) | 147 D |
| [Portex](https://gitcoin.co/grants/2166/portex-secure-crypto-portfolio-management-and-exc) | 147 D |
| [Satellite.im](https://gitcoin.co/grants/1695/satelliteim-a-fast-painless-decentralized-chat) | 1029 D |
| [Voodfy](https://gitcoin.co/grants/2318/decentralized-video-platform-for-creators-to-host) | 1029 D |
| [Wrapped Filecoin](https://gitcoin.co/grants/1454/wrapped-filecoin-wfil) | 735 D |

### Job Referral Program

We have a couple of positions [front-end engineer and go engineer](https://textile.breezy.hr/) that we'd love your help filling! If not you, we also have a paid referral program if you know any talented candidates, [see here](https://app.drafted.us/c/textileio).

**Comment by andrewxhill** [comment]
## Textile Update-  July, 2021

### Bridging Filecoin to a blockchain near you

Our first Filecoin bridge is live! [Read all about it here](https://blog.textile.io/native-filecoin-storage-for-blockchains/). Share [the tweet](https://twitter.com/textileio/status/1412817446868635650) with all your fans too! This is the first part in our development of native Filecoin storage bridges across multiple different blockchains.

The first bridge is available on the NEAR testnet, you can read the docs here, https://near.storage.

### New tooling to improve storage workflows

* We released [_bidbot_](https://github.com/textileio/bidbot), a library for miners to run and connect to an auction feed coming from our blockchain bridge. 
* We released a [batch CAR generation tool](https://docs.textile.io/powergate/offline/#filecoin-batching), that helps you pack many files into a single CAR file suitable for partial retrieval. 

Read more about both releases here: https://blog.textile.io/introducing-storage-auctions-filecoin/

### Coming up!

Tomorrow, we'll publish our technical blog post about using the NEAR~Filecoin bridge.
